### PR TITLE
ED/IP Patient Data

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: app/models/health/ed_ip_visit_file_v1.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# ### HIPAA Risk Assessment
+# Risk: Relates to a patient and contains PHI
+# Control: PHI attributes documented
+module Health
+  class EdIpVisitFileV1 < EdIpVisitFile
+    acts_as_paranoid
```
Comment (eanders) [inline app/models/health/ed_ip_visit_file_v1.rb:12]: Does this need the PHI declarations that were removed from the base file?

File: app/models/health/loaded_ed_ip_visit.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+###
+
+# ### HIPAA Risk Assessment
+# Risk: Relates to a patient and contains PHI
+# Control: PHI attributes documented
+module Health
+  class LoadedEdIpVisit < HealthBase
+    acts_as_paranoid
+
+    phi_patient :medicaid_id
```
Comment (eanders) [inline app/models/health/loaded_ed_ip_visit.rb:14]: Maybe these are them here?

File: db/health/migrate/20211005200728_set_ed_ip_visit_file_type.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class SetEdIpVisitFileType < ActiveRecord::Migration[5.2]
+  def change
```
Comment (eanders) [inline db/health/migrate/20211005200728_set_ed_ip_visit_file_type.rb]: this should probably be `up`

File: db/health/migrate/20211007123413_populate_ed_ip_visits.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class PopulateEdIpVisits < ActiveRecord::Migration[5.2]
+  def change
+    Health::EdIpVisitFile.find_each do |file|
+      file.ingest!(Health::LoadedEdIpVisit.from_file(file))
+    end
+  end
```
Comment (eanders) [inline db/health/migrate/20211007123413_populate_ed_ip_visits.rb]: This feels like maybe it should be a rake task or called manually, not a migration.

File: app/models/health/ed_ip_visit_file_v1.rb (hunk 1 above)
Comment (awisema) [reply app/models/health/ed_ip_visit_file_v1.rb:12]: This is the visit file -- so, it is basically free text, and the PHI declarations are in the base class `EdIpVisitFile`. The per field PHI is in `LoadedEdIpVisit` and `EdIpVisit`.

File: db/health/migrate/20211007123413_populate_ed_ip_visits.rb (hunk 4 above)
Comment (awisema) [reply db/health/migrate/20211007123413_populate_ed_ip_visits.rb]: Replaced with the need to call `Health::EdIpVisit::populate!` after deployment.

**Review by eanders — APPROVED** [review]

**Pull request merged**